### ISSUE TITLE
error registering campaign view

### ISSUE THREAD

**Issue by SKJoy** [knadh/listmonk#1588]
**Version:**
 - ListMonk: **2.5.1** (a6a2b69 2023-08-15T15:49:28Z, linux/amd64)
 - OS: **Docker** in AlmaLinux 8

**Description of the bug and steps to reproduce:**
Logs frequently coming up with the following error messages;
```
2023/11/11 05:32:13error registering campaign view: pq: null value in column "campaign_id" of relation "campaign_views" violates not-null constraint
2023/11/11 05:32:13error registering campaign view: code=500, message=Error updating Campaign: pq: null value in column "campaign_id" of relation "campaign_views" violates not-null constraint. Failing row contains (3101, null, null, 2023-11-11 05:32:13.375611+00).
```

This is happening to another instance of LisMonk on another server.

**Comment by knadh** [comment]
This could be because the views are coming in for a campaign that's been deleted. Could you please check whether that could be the case?

**Comment by SKJoy** [comment]
@knadh Yes I thought that too. Is it not possible to put something under the maintenance to clean up the database from those orphaned records? I tried the maintenance a few times, didn't resolve that VIEW ERROR issue :( Also, it is a bit difficult to inspect inside the database especially while ListMonk is from within Docker.

**Comment by knadh** [comment]
There are no orphan records with campaigns. Once a campaign is deleted, it’s gone for good.

What is happening here is that users are opening email campaigns long after the campaign is deleted. listmonk is then unable to record that view.

I think the right approach here is to suppress this particular error from being logged as it’s an expected scenario. Will push a patch for this.
